pkg/cluster/sync.go

```diff
@@ -658,7 +658,9 @@ func (c *Cluster) syncStatefulSet() error {
 	// if we get here we also need to re-create the pods (either leftovers from the old
 	// statefulset or those that got their configuration from the outdated statefulset)
 	if len(podsToRecreate) > 0 {
-		if isSafeToRecreatePods {
+		if !isInMaintenanceWindow(c.Spec.MaintenanceWindows) {
+			c.logger.Infof("postpone pod recreation - not in maintenance window")
+		} else if isSafeToRecreatePods {
```

Comment:
just a refactoring question - maybe !isInMaintenanceWindow could be it's own if a few lines above to set isSafeToRecreatePods to false. We could collect the reason from all the blocks that set isSafeToRecreatePods = false and then we log here: c.logger.Warnf("postpone pod recreation - reason: %s", strings.Joins(postponeReasons))